php/php.blade/src/org/netbeans/modules/php/blade/csl/elements/PhpFunctionElement.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.blade.csl.elements;
+
+import java.util.List;
+import org.netbeans.modules.csl.api.ElementKind;
+import org.openide.filesystems.FileObject;
+
+/**
+ *
+ * @author bhaidu
+ */
+public class PhpFunctionElement extends NamedElement {
+
+    public final List<String> params;
+    public final String namespace;
+
+    public PhpFunctionElement(String name, FileObject file,
+            ElementType type,
+            String namespace,
+            List<String> params) {
+        super(name, file, type);
+        this.namespace = namespace;
+        this.params = params;
```

Comment:
```suggestion
        this.params.addAll(params);
```

Reply:
@mbien Yes, I suggested `private final List<String> params = new ArrayList<>();` above but it has been ignored yet...

Reply:
sorry I overlooked that! Buf if there are no null-elements, `List.copyOf()` would be still potentially better since it is optimized for immutable data.

Reply:
Yes, it's better.(`params` is not modified at all in this class.) Anyway, we would like to avoid `this.params = params;`.